### ISSUE TITLE
Index was out of range error for large number of rows in arabic language (rtl), after upgrading from version 0.96 to 0.97

### ISSUE THREAD

**Issue by rubeesh** [ClosedXML/ClosedXML#1916]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Version of ClosedXML**

e.g. 0.97.0

Index was out of range error, if the data has large number of rows in arabic language (rtl)  at line ws.Columns().AdjustToContents . 
The issue started with version 0.97.  It was working fine with 0.96 .  

```
 using var wb = new XLWorkbook();
 var ws = wb.Worksheets.Add();
 ws.RightToLeft = true;

.....
codes removed
....

 ws.Cell(3, 1).InsertData(items);
.....
codes removed
....
 ws.Columns().AdjustToContents(); // Error
```
Error:
![Screenshot 2022-11-18 093240](https://user-images.githubusercontent.com/26519001/202637919-d5daedcc-90c0-49cf-a7e9-3fbe4db10c1f.png)

**Comment by bdariusb** [comment]
same issue, don't think it's related to language, but row number was 60k for our case

**Comment by rubeesh** [comment]
It is working with English, the issue is with only Arabic language. 
Moreover it was working fine with 0.96.0

**Comment by rubeesh** [comment]
Any update on this issue

**Comment by jahav** [comment]
@rubeesh No. This issue doesn't even have reproducible scenario and exception is shown as a screenshot without a stacktrace.

I have tried this and no exception happened.
```csharp
            using var wb = new XLWorkbook();
            var ws = wb.Worksheets.Add();
            ws.RightToLeft = true;

            var items = Enumerable.Range(0, 60000)
                .Select(x => "الافترايفكتس و البريمير و الافد ميدا كومبوزر و السموك و برامج اخرى كثيرة").ToList();
            ws.Cell(3, 1).InsertData(items);
            ws.Columns().AdjustToContents();
```

**Comment by rubeesh** [comment]
@jahav 
**Please find the below code which could reproduce error:**
```
            using var wb = new XLWorkbook();
            var ws = wb.Worksheets.Add();
            ws.RightToLeft = true;

             var items = Enumerable.Range(0, 1000).Select(x => new{
                                    H1 = "الهند وإنجلترا ومصر وأستراليا وباكستان والصينالهند وإنجلترا ومصر وأستراليا وباكستان والصين ",
                                    H2 = "اليا وباكستان والصين",
                                    H3 = "كستان والصين",
                                    H4 = "الهند ولصين",
                                    H5 = "الهند وإنجلترا ومصر وأستراليا وباكستان والصين",
                                    H6 = "الهند وإاليا وباكستان والصين",
                                    H7 = "اوالصين",
                                    H8 = 63,
                                    H9 = "الهند وإنجلتراوالصين",
                                    H10 = 123,
                                    H11 = "الهند ",
                                    H12 = "الاكستان والصين",
                                    H13 = "اصين",
                                    H14 = 999,
                                    H15 = 569,
                                    H16 = "الهن",
                                    H17 = "الهند وإنجلترا ومصر وأستراليا وباكستان والصين",
                                    H18 = "الهند وإنجلترا ومصر وأستراليا وباكستان والصين"
                                }).ToList();
            ws.Cell(3, 1).InsertData(items);

            ws.Columns().AdjustToContents();  // Error line
```

**Error:**  
`{"Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')"}`

**Stacktrace:**
```
"  at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException()\r\n  
   at SixLabors.Fonts.GlyphSubstitutionCollection.GetGlyphShapingData(Int32 index)\r\n  
   at SixLabors.Fonts.Tables.AdvancedTypographic.AdvancedTypographicUtils.Match[T](Int32 increment, T[] sequence, SkippingGlyphIterator iterator, Func`3 condition, Span`1 matches)\r\n  
   at SixLabors.Fonts.Tables.AdvancedTypographic.AdvancedTypographicUtils.MatchCoverageSequence(SkippingGlyphIterator iterator, CoverageTable[] coverageTable, Int32 increment)\r\n  
   at SixLabors.Fonts.Tables.AdvancedTypographic.AdvancedTypographicUtils.CheckAllCoverages(FontMetrics fontMetrics, LookupFlags lookupFlags, IGlyphShapingCollection collection, Int32 index, Int32 count, CoverageTable[] input, CoverageTable[] backtrack, CoverageTable[] lookahead)\r\n  
   at SixLabors.Fonts.Tables.AdvancedTypographic.GSub.LookupType6Format3SubTable.TrySubstitution(FontMetrics fontMetrics, GSubTable table, GlyphSubstitutionCollection collection, Tag feature, Int32 index, Int32 count)\r\n  
   at SixLabors.Fonts.Tables.AdvancedTypographic.GSubTable.ApplySubstitution(FontMetrics fontMetrics, GlyphSubstitutionCollection collection)\r\n  
   at SixLabors.Fonts.TextLayout.DoFontRun(ReadOnlySpan`1 text, Int32 start, IReadOnlyList`1 textRuns, Int32& textRunIndex, Int32& codePointIndex, Int32& bidiRunIndex, Boolean isFallbackRun, Font font, BidiRun[] bidiRuns, Dictionary`2 bidiMap, GlyphSubstitutionCollection substitutions, GlyphPositioningCollection positionings)\r\n  
   at SixLabors.Fonts.TextLayout.ProcessText(ReadOnlySpan`1 text, TextOptions options)\r\n  
   at ClosedXML.Graphics.DefaultGraphicEngine.GetTextWidth(String text, IXLFontBase fontBase, Double dpiX)\r\n  
   at ClosedXML.Excel.XLColumn.GetWidthInCharacters(String text, IXLFontBase font)\r\n  
   at ClosedXML.Excel.XLColumn.AdjustToContents(Int32 startRow, Int32 endRow, Double minWidth, Double maxWidth)\r\n  
   at ClosedXML.Excel.XLColumns.<>c.<AdjustToContents>b__13_0(XLColumn c)\r\n  
   at ClosedXML.Excel.EnumerableExtensions.ForEach[T](IEnumerable`1 source, Action`1 action)\r\n  
   at ClosedXML.Excel.XLColumns.AdjustToContents()\r\n  
   at Abc.Server.Services.DownloadService.<GenerateExcel>d__6`1.MoveNext() in C:\\Users\\Abc\\source\\repos\\Abc\\Server\\Services\\DownloadService.cs:line 133"
```
**Error only for Arabic language and verion 0.97 and above. It was working well with version 0.96**

**Comment by rubeesh** [comment]
Any update on this issue

**Comment by jahav** [comment]
@rubeesh I checked what exactly caused the problem and found a single text was enough to cause exception. I then checked the SixFonts library. Someone already reported it https://github.com/SixLabors/Fonts/issues/302 and it is fixed in beta19. I will update dependency.

In the meantime, add a direct dependency to your project for SixLabors.Fonts beta19 and it should use the beta19 through [direct-dependency-wins](https://learn.microsoft.com/en-us/nuget/concepts/dependency-resolution#direct-dependency-wins). That should solve the problem (except horrible slowness, but that is another issue to be solved).

```csharp
	var fontFamily = SystemFonts.Get("Calibri");
	var font = fontFamily.CreateFont(16);

	TextMeasurer.Measure("اصين", new TextOptions(font));
```
